apis/kueue/v1alpha2/clusterqueue_types.go

```diff
@@ -243,10 +243,63 @@ type ClusterQueueStatus struct {
 	// clusterQueue and haven't finished yet.
 	// +optional
 	AdmittedWorkloads int32 `json:"admittedWorkloads"`
+
+	// conditions hold the latest available observations of the ClusterQueue
+	// current state.
+	// +optional
+	// +listType=map
+	// +listMapKey=type
+	Conditions []ClusterQueueCondition `json:"conditions,omitempty"`
 }
 
 type UsedResources map[corev1.ResourceName]map[string]Usage
 
+type ClusterQueueCondition struct {
+	// type of condition could be:
+	//
+	// ClusterQueuePending: clusterQueue is accepted but not yet active.
+	// In this state, it can't admit new workloads and its quota can't be borrowed
+	// by other active ClusterQueues in the same cohort.
+	//
+	// ClusterQueueActive: clusterQueue can admit new workloads and its quota
+	// can be borrowed by other ClusterQueues in the same cohort.
+	Type ClusterQueueConditionType `json:"type"`
+
+	// status could be True, False or Unknown.
+	Status corev1.ConditionStatus `json:"status"`
+
+	// lastProbeTime is the last time the condition was checked.
+	// +optional
+	LastProbeTime metav1.Time `json:"lastProbeTime,omitempty"`
+
+	// lastTransitionTime is the last time the condition transit from one status
+	// to another.
+	// +optional
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty"`
+
+	// reason is a brief reason for the condition's last transition.
+	// +optional
+	Reason string `json:"reason,omitempty"`
+
+	// message is a human-readable message indicating details about last
+	// transition.
+	// +optional
+	Message string `json:"message,omitempty"`
+}
+
+type ClusterQueueConditionType string
+
+const (
+	// ClusterQueuePending indicates clusterQueue is accepted but not yet active.
+	// In this state, it can't admit new workloads and its quota can't be borrowed
+	// by other active ClusterQueues in the same cohort.
+	ClusterQueuePending ClusterQueueConditionType = "Pending"
+
+	// ClusterQueueActive indicates clusterQueue can admit new workloads and its quota
+	// can be borrowed by other ClusterQueues in the same cohort.
```

Comment:
@kerthcet did we implement the behavior that an inactive ClusterQueue's quota can't be borrowed?

Reply:
Yes, we'll filter out the inactive cq in snapshotting https://github.com/kubernetes-sigs/kueue/blob/d136275b81b9592e7a8c888d6f5ed8af2551ce59/pkg/cache/snapshot.go#L54-L61